src/app/maptools/qgsmaptoolshapecircle2tangentspoint.cpp

```diff
@@ -101,6 +101,12 @@ bool QgsMapToolShapeCircle2TangentsPoint::cadCanvasReleaseEvent( QgsMapMouseEven
   }
   else if ( e->button() == Qt::RightButton )
   {
+    if ( mPoints.size() < 4 || mCircle.isEmpty() )
+    {
+      clean();
```

Comment:
Does this mean if I right click after picking two points then those points are lost? I think ideally it'd just ignore the right click in that situation, not drop the existing points.

Reply:
Right clicking _before_ we have enough points from the user means _cancel_ in other digitizing tools too.
I'd consider this the same as _create polygon_ with two vertices, or _create line_ with a single vertex.
Why treat this differently?

Reply:
It's just a little unpredictable how these tools work -- as you've noted you have to left click a bunch of times, and then at some stage the left clicks start being ignored and you have to THEN do a right click. So if we abort when a right click happens before that magic threshold of left clicks is reached then it's really un user-friendly.

I think ideally the tools would give some indication of when the feature is valid and ready to be saved. Eg the rubber band becomes green or something at that stage...

Reply:
I'd insist that in order to have predictable tools the right click should _always_ result in either:
- Finish digitizing _if enough points are collected_
- Restart digitizing _if not enough points are collected_

This is how the main digitizing tools work, so those weird cousins should follow form.

Let's address the clunky ux in another pr. I'd suggest instead of coloring the rubberband, we add one more left click step that freezes the rubberband (ok, maybe change its color too) marking it as _ready to finish_. At this point the mouse does not update the rubberband any more and the user may either:
- right click to finish
- esc to reset
- backspace/del to unfreeze/pick another last point